readthedocs/vcs_support/backends/git.py

```diff
@@ -35,10 +35,20 @@ class Backend(BaseVCS):
     repo_depth = 50
 
     def __init__(self, *args, **kwargs):
+        # The version_identifier is a Version.identifier value passed from the build process.
+        # It has a special meaning since it's unfortunately not consistent, you need to be aware of
+        # exactly how and where to use this.
+        # See more in the .get_remote_fetch_refspec() docstring
+        self.version_identifier = kwargs.pop("version_identifier")
+        # We also need to know about Version.machine
+        self.version_machine = kwargs.pop("version_machine")
```

Comment:
Looks like we are passing these to all vcs backends

https://github.com/readthedocs/readthedocs.org/blob/2f3e6ca3667c58042acec96fcab8fee181e8eb32/readthedocs/projects/models.py#L1022-L1030C13

this will break other VCS, we should move these attributes to the base class.

Reply:
The base class ignores unknown kwargs: https://github.com/readthedocs/readthedocs.org/blob/524c24ca38b429e9b3959e5811d34379326b9159/readthedocs/vcs_support/base.py#L64